mne/channels/layout.py

```diff
@@ -927,6 +933,10 @@ def apply_montage(info, montage):
         info['chs'][ch_idx]['loc'] = np.r_[pos, [0.] * 9]
         sensors_found = True
 
+        if ch_type == 'ieeg':
+            info['chs'][ch_idx]['kind'] = FIFF.FIFFV_SEEG_CH
+            info['chs'][ch_idx]['coord_frame'] = FIFF.FIFFV_COORD_MRI
```

Comment:
you're mixing objectives here. we have another function to change channel types. See rename_channels function.